src/test/java/rx/internal/operators/OnSubscribeCollectTest.java

```diff
@@ -0,0 +1,253 @@
+            }
+            
+        }, new Action2<StringBuilder, Integer>() {
+
+            @Override
+            public void call(StringBuilder sb, Integer v) {
+                if (sb.length() > 0) {
+                    sb.append("-");
+                }
+                sb.append(v);
+            }
+        }).toBlocking().last().toString();
+
+        assertEquals("1-2-3", value);
+    }
+    
+    @Test
+    public void testFactoryFailureResultsInErrorEmission() {
+        TestSubscriber<Object> ts = TestSubscriber.create();
+        final RuntimeException e = new RuntimeException();
+        Observable.just(1).collect(new Func0<List<Integer>>() {
+
+            @Override
+            public List<Integer> call() {
+                throw e;
+            }
+        }, new Action2<List<Integer>, Integer>() {
+
+            @Override
+            public void call(List<Integer> list, Integer t) {
+                list.add(t);
+            }
+        }).subscribe(ts);
+        ts.assertNoValues();
+        ts.assertError(e);
+        ts.assertNotCompleted();
+    }
+    
+    @Test
+    public void testCollectorFailureDoesNotResultInTwoErrorEmissions() {
+        try {
+            final List<Throwable> list = new CopyOnWriteArrayList<Throwable>();
+            RxJavaHooks.setOnError(new Action1<Throwable>() {
+
+                @Override
+                public void call(Throwable t) {
+                    list.add(t);
+                }
+            });
+            final RuntimeException e1 = new RuntimeException();
+            final RuntimeException e2 = new RuntimeException();
+            TestSubscriber<List<Integer>> ts = TestSubscriber.create();
+            Observable.create(new OnSubscribe<Integer>() {
+
+                @Override
+                public void call(final Subscriber<? super Integer> sub) {
+                    sub.setProducer(new Producer() {
+
+                        @Override
+                        public void request(long n) {
+                            if (n > 0) {
+                                sub.onNext(1);
+                                sub.onError(e2);
+                            }
+                        }
+                    });
+                }
+            }).collect(new Func0<List<Integer>>() {
+
+                @Override
+                public List<Integer> call() {
+                    return new ArrayList<Integer>();
+                }
+            }, //
+                    new Action2<List<Integer>, Integer>() {
+
+                        @Override
+                        public void call(List<Integer> t1, Integer t2) {
+                            throw e1;
+                        }
+                    }).unsafeSubscribe(ts);
+            assertEquals(Arrays.asList(e1), ts.getOnErrorEvents());
+            ts.assertNotCompleted();
+            assertEquals(Arrays.asList(e2), list);
+        } finally {
+            RxJavaHooks.setOnError(null);
```

Comment:
You have to call `reset` otherwise RxJavaHooks wont delegate back to `RxJavaPlugins`.